frontend/src/metabase/dashboard/hoc/DashboardData.jsx

```diff
@@ -56,7 +56,9 @@ export default ComposedComponent =>
         initialize({ clear: !isNavigatingBackToDashboard });
 
         try {
-          await fetchDashboard(dashboardId, location && location.query);
+          await fetchDashboard(dashboardId, location && location.query, {
+            preserveDashboard: isNavigatingBackToDashboard,
```

Comment:
Based on my comment here (TODO: add a link to the comment in `frontend/src/metabase/dashboard/actions/data-fetching.js`), you can see how we're using `isNavigatingBackToDashboard` differently here, one without negation and one with negation, which could be mildly confusing.